src/main/java/foundation/stack/jdbc/ConnectionLookupRegistry.java

```diff
@@ -27,10 +32,16 @@ public void deregisterLookup(ConnectionLookup lookup) {
     }
 
     public ConnectionLookupResult lookupConnection(String query) {
+        ConnectionLookupResult connectionLookupResult = connectionLookupResults.get(query);
```

Comment:
Is this called often? If we want to cache, we should use a guava cache so it potentially gets cleaned up.